### PR TITLE
Move functions to separate files in context package 

### PR DESCRIPTION
**Pull request by adisky**
**What type of PR is this?**

/kind code-refactoring

**What does does this PR do / why we need it**:

part of context refactoring

**Which issue(s) this PR fixes**:

Partially : https://github.com/openshift/odo/issues/4081

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
Tests should pass

### CODE REVIEW

File: pkg/odo/genericclioptions/resolve.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	"fmt"
+
+	"github.com/openshift/odo/pkg/envinfo"
+	"github.com/openshift/odo/pkg/odo/util"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ResolveAppFlag resolves the app from the flag
+func ResolveAppFlag(command *cobra.Command) string {
```
Comment (girishramnani) [inline pkg/odo/genericclioptions/resolve.go:13]: Doesn’t this make more sense to be in flag.go

Comment (adisky) [reply pkg/odo/genericclioptions/resolve.go:13]: i will try to remove this in next PR's

Comment (girishramnani) [reply pkg/odo/genericclioptions/resolve.go:13]: Ok

**Comment by girishramnani** [comment]
/approve
/lgtm 

This refactor PR pass the tests

**Pull request merged**